ui/app/components/auth-form.js

```diff
@@ -284,25 +283,24 @@ export default Component.extend(DEFAULTS, {
   }),
 
   actions: {
-    doSubmit(passedData, event) {
+    doSubmit(passedData, event, token) {
       if (event) {
         event.preventDefault();
       }
-      let data = {};
-      this.setProperties({
-        error: null,
-      });
-      // if callback from oidc we have a token at this point
-      let backend =
-        this.providerName === 'oidc' ? this.getAuthBackend('token') : this.selectedAuthBackend || {};
-      let backendMeta = BACKENDS.find(
+      if (token) {
+        this.set('token', token);
+      }
+      this.set('error', null);
+      // if callback from oidc or jwt we have a token at this point
+      const backend = token ? this.getAuthBackend('token') : this.selectedAuthBackend || {};
+      const backendMeta = BACKENDS.find(
         (b) => (b.type || '').toLowerCase() === (backend.type || '').toLowerCase()
       );
-      let attributes = (backendMeta || {}).formAttributes || [];
+      const attributes = (backendMeta || {}).formAttributes || [];
+      const data = this.getProperties(...attributes);
 
-      data = assign(data, this.getProperties(...attributes));
       if (passedData) {
-        data = assign(data, passedData);
+        Object.assign(data, passedData);
```

Comment:
nice, this reads easier too

Reply:
That `assign` polyfill is deprecated so we'll have to replace all of them at some point. I think it's good until version 5.